hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -155,60 +155,66 @@ mkChain tracer queryTimeHandle wallet@TinyWallet{getUTxO} ctx LocalChainState{ge
         timeHandle <- queryTimeHandle
         vtx <-
           atomically (prepareTxToPost timeHandle wallet ctx spendableUTxO tx)
-            >>= finalizeTx wallet ctx spendableUTxO mempty
+            >>= finalizeTx tracer wallet ctx spendableUTxO mempty
         submitTx vtx
     , -- Handle that creates a draft commit tx using the user utxo.
       -- Possible errors are handled at the api server level.
-      draftCommitTx = \headId utxoToCommit -> do
+      draftCommitTx = \headId utxoToCommit blueprintTx -> do
         ChainStateAt{spendableUTxO} <- atomically getLatest
         walletUtxos <- atomically getUTxO
         let walletTxIns = fromLedgerTxIn <$> Map.keys walletUtxos
         let userTxIns = Set.toList $ UTxO.inputSet utxoToCommit
         let matchedWalletUtxo = filter (`elem` walletTxIns) userTxIns
         -- prevent trying to spend internal wallet's utxo
         if null matchedWalletUtxo
-          then
-            traverse (finalizeTx wallet ctx spendableUTxO (fst <$> utxoToCommit)) $
-              commit' ctx headId spendableUTxO utxoToCommit
+          then do
+            traverse (finalizeTx tracer wallet ctx spendableUTxO utxoToCommit) $
+              commit ctx headId spendableUTxO utxoToCommit blueprintTx
           else pure $ Left SpendingNodeUtxoForbidden
     , -- Submit a cardano transaction to the cardano-node using the
       -- LocalTxSubmission protocol.
       submitTx
     }
 
 -- | Balance and sign the given partial transaction.
+-- TODO: remove tracing
 finalizeTx ::
+  Tracer m DirectChainLog ->
   MonadThrow m =>
   TinyWallet m ->
   ChainContext ->
   UTxO.UTxO ->
   UTxO.UTxO ->
   Tx ->
   m Tx
-finalizeTx TinyWallet{sign, coverFee} ctx utxo userUTxO partialTx = do
+finalizeTx tracer TinyWallet{sign, coverFee} ctx utxo userUTxO partialTx = do
   let headUTxO = getKnownUTxO ctx <> utxo <> userUTxO
   coverFee headUTxO partialTx >>= \case
     Left ErrNoFuelUTxOFound ->
-      throwIO (NoFuelUTXOFound :: PostTxError Tx)
+      traceWith tracer (BlahBlah "ErrNoFuelUTxOFound")
+        >> throwIO (NoFuelUTXOFound :: PostTxError Tx)
```

Comment:
Why not use `trace "ErrNoFuelUTxOFound" throwIO (NoFuelUTXOFound :: PostTxError Tx)` for temporary tracing?